celltype_annotation/analysis/03-cell-assign-combined-markers.Rmd

```diff
@@ -0,0 +1,392 @@
+custom_marker_genes <- panglao_df |> 
+  dplyr::filter(organ %in% muscle_cells |
+                `cell type` == "Monocytes")
+
+custom_marker_genes |>
+  dplyr::count(`cell type`, `organ`)
+```
+
+```{r}
+# list of matrix files to create
+mtx_files <- c(panglao_comb_mtx_file,
+               panglao_custom_mtx_file)
+
+# if any of them don't exist, make them 
+if(!any(file.exists(mtx_files))){
+ 
+  # create rowdata df
+  rowdata_df <- rowData(processed_sce) |>
+    as.data.frame() |>
+    tibble::rownames_to_column("ensembl_id") |>
+    dplyr::select(ensembl_id, gene_symbol)
+  
+  # create list of marker gene sets
+  all_marker_genes <- list(panglao_tissue_immune_combined,
+                           custom_marker_genes)
+  names(all_marker_genes) <- c("combined", "combined_custom")
+  
+  # get binary mtx for combined tissues 
+  binary_matrix_list <- all_marker_genes |> 
+    purrr::map(\(gene_list) build_binary_mtx(marker_genes_df = gene_list,
+                                             celltype_column = 'cell type',
+                                             gene_id_column = 'official gene symbol',
+                                             rowdata_df = rowdata_df))
+  
+  # export mtx 
+  purrr::walk2(binary_matrix_list, mtx_files,
+               \(binary_mtx, file)
+               readr::write_csv(binary_mtx, file))
+  
+}
+```
+
+```{r}
+# list of output prediction files 
+prediction_files <- c(combined_prediction_file,
+                      custom_prediction_file) 
+```
+
+
+```{r,  eval = FALSE}
+# run cell assign with muscle ref
+cellassign_calls <- purrr::map2(prediction_files, mtx_files,
+                                \(predictions, marker_genes) {
+                                  glue::glue("python ../scripts/cell-assign.py \
+                                              --input_anndata '{anndata_file}' \
+                                              --output_predictions '{predictions}' \
+                                              --reference '{marker_genes}'")
+                                })
+
+system(cellassign_call)
+```
+
+## CellAssign Results
+
+```{r}
+# add muscle only to prediction file list 
+prediction_files <- c(prediction_files, muscle_only_prediction_file)
+names(prediction_files) <- c("combined", "combined_custom", "muscle_only")
+
+# read in prediction scores 
+all_predictions <- prediction_files |>
+  purrr::map(readr::read_tsv)
+```
+
+```{r}
+# get final cell type assignments 
+all_assignments <- purrr::map(all_predictions,
+                              get_celltype_assignments)
+```
+
+
+```{r}
+# print table of assignments
+all_assignments |>
+  purrr::map(\(celltype_assignments){
+    
+    # print out table of assignments
+    celltype_assignments |> 
+      dplyr::count(celltype) |>
+      dplyr::arrange(desc(n))
+    
+  })
+```
+
+```{r, fig.height=5, fig.width=10}
+# heatmap of prediction scores 
+all_predictions |>
+  purrr::imap(\(prediction_mtx, organ_type){
```

Comment:
Something is funky here.. the last heatmap is overlayed with a second `combined_custom` heatmap. I played around with this for a bit, and the last matrix is always overlayed with the previous one. Changing to `iwalk` seems to do the trick!

```suggestion
  purrr::iwalk(\(prediction_mtx, organ_type){
```


<img width="568" alt="Screenshot 2023-07-19 at 11 20 02 AM" src="https://github.com/AlexsLemonade/sc-data-integration/assets/4701111/189618a8-c243-4961-9329-ee47a3ab2015">